dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -0,0 +1,398 @@
+                         " and rightType do no match on entityType");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        if (!verifyMaxCardinality(context, relationship.getLeftItem(),
+                                  relationshipType.getLeftMaxCardinality(), relationshipType)) {
+            log.warn("The relationship has been deemed invalid since the left item has more" +
+                         " relationships than the left max cardinality allows after we'd store this relationship");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        if (!verifyMaxCardinality(context, relationship.getRightItem(),
+                                  relationshipType.getRightMaxCardinality(), relationshipType)) {
+            log.warn("The relationship has been deemed invalid since the right item has more" +
+                         " relationships than the right max cardinality allows after we'd store this relationship");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        return true;
+    }
+
+    private void logRelationshipTypeDetails(RelationshipType relationshipType) {
+        log.warn("The relationshipType's ID is: " + relationshipType.getID());
+        log.warn("The relationshipType's left label is: " + relationshipType.getLeftLabel());
+        log.warn("The relationshipType's right label is: " + relationshipType.getRightLabel());
+        log.warn("The relationshipType's left entityType label is: " + relationshipType.getLeftType().getLabel());
+        log.warn("The relationshipType's right entityType label is: " + relationshipType.getRightType().getLabel());
+        log.warn("The relationshipType's left min cardinality is: " + relationshipType.getLeftMinCardinality());
+        log.warn("The relationshipType's left max cardinality is: " + relationshipType.getLeftMaxCardinality());
+        log.warn("The relationshipType's right min cardinality is: " + relationshipType.getRightMinCardinality());
+        log.warn("The relationshipType's right max cardinality is: " + relationshipType.getRightMaxCardinality());
+    }
+
+    private boolean verifyMaxCardinality(Context context, Item itemToProcess,
+                                         int maxCardinality, RelationshipType relationshipType) throws SQLException {
+        List<Relationship> rightRelationships = findByItemAndRelationshipType(context, itemToProcess, relationshipType,
+                                                                              false);
+        if (rightRelationships.size() >= maxCardinality && maxCardinality != 0) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean verifyEntityTypes(Item itemToProcess, EntityType entityTypeToProcess) {
+        List<MetadataValue> list = itemService.getMetadata(itemToProcess, "relationship", "type", null, Item.ANY);
```

Comment:
The type field to be used is currently not decided yet. Since it doesn't impact the functionality, that discussion is currently lower priority. Once the final field has been decided, it will be moved to the MetadataSchemaEnum